src/wp-includes/class-wp-query.php

```diff
@@ -3529,7 +3530,12 @@ public function get_posts() {
 			$this->posts = array_map( 'get_post', $this->posts );
 
 			if ( $q['cache_results'] ) {
-				update_post_caches( $this->posts, $post_type, $q['update_post_term_cache'], $q['update_post_meta_cache'] );
+				if ( $unfiltered_query ) {
```

Comment:
Per my original feedback in https://github.com/WordPress/wordpress-develop/pull/5134#discussion_r1330566960: We also have to ensure the query results were not modified, by storing the value from https://github.com/WordPress/wordpress-develop/blob/3bacfc9f978876549ee0e2b5068be02e006d956c/src/wp-includes/class-wp-query.php#L3337 in a variable, e.g. `$unfiltered_posts`.

Then we can check it here like this:
```suggestion
				if ( $unfiltered_query && $this->posts === $unfiltered_posts ) {
```

Reply:
I see now. Good catch. Made the tweak https://github.com/WordPress/wordpress-develop/pull/5251/commits/f575f45ae1505271eaa59d5753f10a7269917c4a.